habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/util/ContainerizeDepsDockerfileHelper.java

```
@@ -37,158 +35,215 @@ public ContainerizeDepsDockerfileHelper(ContainerizeDepsMojo containerizeDepsMoj
+    private String getBuilderStageStartComment() {
+        return HABUSHU_BUILDER_STAGE + HABUSHU_COMMENT_START;
+    }
+
+    private String getBuilderStageEndComment() {
+        return HABUSHU_BUILDER_STAGE + HABUSHU_COMMENT_END;
+    }
+
+    private String getFinalStageStartComment() {
+        return HABUSHU_FINAL_STAGE + HABUSHU_COMMENT_START;
+    }
+
+    private String getFinalStageEndComment() {
+        return HABUSHU_FINAL_STAGE + HABUSHU_COMMENT_END;
+    }
+
+    private String getWrappedBuilderStageContent(){
+        return getWrappedBuilderStageContent(false);
+    }
+
+    private String getWrappedBuilderStageContent(Boolean includeNewLineAtEnd) {
+        String builderStageContent = getBuilderStageContent();
+        String wrappedBuilderStageContent = getBuilderStageStartComment() + "\n" + builderStageContent + "\n" + getBuilderStageEndComment();
+        if (includeNewLineAtEnd) {
+            return wrappedBuilderStageContent + "\n";
+        }
+        return wrappedBuilderStageContent;
+    }
+
+    private String getWrappedFinalStageContent(){
+        return getWrappedFinalStageContent(false);
+    }
+
+    private String getWrappedFinalStageContent(Boolean includeNewLineAtEnd) {
+        String finalStageContent = getFinalStageContent();
+        String wrappedFinalStageContent = getFinalStageStartComment() + "\n" + finalStageContent + "\n" + getFinalStageEndComment();
+        if (includeNewLineAtEnd) {
+            return wrappedFinalStageContent + "\n";
+        }
+        return wrappedFinalStageContent;
+    }
+
+    private String updateDockerfileWithBuilderLogic() {
+        try {
+            String dockerfileContent = Files.readString(Paths.get(containerizeDepsMojo.getDockerfile().toString()));
+            if (!builderContentPresent(dockerfileContent)) {
+                if (!addBuilderContentCommentPresent(dockerfileContent)) {
+                    return getWrappedBuilderStageContent(true) + "\n" + dockerfileContent;
```

Q: Should we raise an error here instead since we don't know what the dockerfileContent looks like?